auslib/web/public/api.yml

```diff
@@ -587,3 +587,210 @@ paths:
       responses:
         200:
           description: version.json content.
+
+# Public Rules endpoints
+  /rules:
+    get:
+      operationId: "auslib.web.public.rules.get"
+      description: ""
+      produces: ["application/json"]
+      responses:
+        200:
+          description: ""
+          schema:
+            $ref: "#/definitions/RulesWrapper"
+
+  /rules/{id_or_alias}:
+    get:
+      operationId: "auslib.web.public.rules.get_by_id_or_alias"
+      description: ""
+      produces: ["application/json"]
+      parameters:
+        - name: id_or_alias
+          in: path
+          required: true
+          type: string
+      responses:
+        200:
+          description: ""
+          schema:
+            $ref: "#/definitions/Rule"
+
+  /rules/{rule_id}/revisions:
+    get:
+      operationId: "auslib.web.public.rules.get_revisions"
+      description: ""
+      produces: ["application/json"]
+      parameters:
+        - name: rule_id
+          in: path
+          required: true
+          type: integer
+      responses:
+        200:
+          description: ""
+          schema:
+            $ref: "#/definitions/RuleHistoryWrapper"
+
+  /releases:
+    get:
+      operationId: "auslib.web.public.releases.get"
+      description: ""
+      produces: ["application/json"]
+      parameters:
+        - name: product
+          in: query
+          required: false
+          type: string
+        - name: name_prefix
+          in: query
+          required: false
+          type: string
+        - name: names_only
+          in: query
+          required: false
+          type: string
+      responses:
+        200:
+          description: ""
+          schema:
+            $ref: "#/definitions/RulesWrapper"
+
+  /releases/{release}:
+    get:
+      operationId: "auslib.web.public.releases.get_release"
+      description: ""
+      produces: ["application/json"]
+      parameters:
+        - name: release
+          in: path
+          required: true
+          type: string
+        - name: pretty
+          in: query
+          required: false
+          type: integer
+      responses:
+        200:
+          description: "Release Blob"
+
+  /releases/{release}/revisions:
+    get:
+      operationId: "auslib.web.public.releases.get_revisions"
+      description: ""
+      produces: ["application/json"]
+      parameters:
+        - name: release
+          in: path
+          required: true
+          type: string
+      responses:
+        200:
+          description: "Release History"
+
+definitions:
```

Comment:
AFAIK the model definitions cannot be shared between files even within the same app in connexion. So if we have two yaml spec files within the app, they are independent of each other. Although jsonschema has defined external references to models to be declared in other files according to jsonschema-draft4 but those references need to be parsed/build explicitly by the program before loading up the swagger.json . Connexion does not do any of that right now.